chainer/variable.py

```diff
@@ -594,6 +598,16 @@ def initialize(self, shape):
         self._data[0] = data
         self.grad = grad
 
+    def update(self):
+        """Updates the data array using the gradient and the update rule.
+
+        This method updates the variable using the update rule attached to this
+        variable. If the gradient is ``None``, it does nothing.
+
+        """
+        if self.update_rule is not None and self._grad is not None:
+            self.update_rule.update(self)
+
```

Comment:
That makes sense. I will fix this part as well as each update rule implementation.